### PR TITLE
Let tot query lastbuild from gcs for a new job

### PR DESCRIPTION
**Pull request by krzyzacy**

### CODE REVIEW

**Comment by krzyzacy** [comment]
ref #2996 - not testing gsutil properly yet

File: prow/cmd/tot/main.go [hunk 1]
```diff
@@ -79,7 +80,19 @@ func (s *store) save() error {
 func (s *store) vend(b string) int {
 	s.mutex.Lock()
 	defer s.mutex.Unlock()
-	n := s.Number[b] + 1
+	n, ok := s.Number[b]
+	if !ok {
+		if out, err := exec.Command("gsutil", "cat", fmt.Sprintf("gs://kubernetes-jenkins/logs/%s/latest-build.txt", b)).Output(); err != nil {
```
Comment (spiffxp) [inline prow/cmd/tot/main.go]: could we gate this behind a flag? or if nothing else, make the bucket uri configurable

Comment (krzyzacy) [inline prow/cmd/tot/main.go]: yeah I'll make the bucket path customizable

**Comment by spxtr** [comment]
I don't think this belongs in prow at all. I really just want to get rid of tot entirely, or else store the build numbers somewhere that works for everyone else.

**Comment by krzyzacy** [comment]
@spxtr after a second thought maybe the whole BuildId thing should not in prow at all. How about in bootstrap?

**Comment by spxtr** [comment]
That makes sense to me.

**Comment by ixdy** [comment]
bootstrap will still need to talk to some service (such as tot) to find the next build ID, since multiple builds of the same job might be running at once.

can we somehow insert a build ID into tot's database when we migrate jobs?

**Comment by krzyzacy** [comment]
@ixdy gsutil grab latestBuild, bump, and then write it back?

**Comment by spxtr** [comment]
Oh, I remember what the issue is. Prow needs to link to gubernator, which uses the build number in the URL. We can work around this by having `bootstrap.py` upload a file pointing from the UUID that prow generates to the actual build number.

**Comment by ixdy** [comment]
I guess `bootstrap.py` can save the latest build number in a file on GCS, and then [use preconditions](https://cloud.google.com/storage/docs/generations-preconditions#rmw-scenario) to increment it atomically?

**Pull request closed without merging**